torch/csrc/jit/python/init.cpp

```diff
@@ -1846,6 +1846,43 @@ void initJITBindings(PyObject* module) {
                 return nullptr;
               }),
           py::call_guard<py::gil_scoped_release>());
+
+  py::class_<PythonAwaitWrapper, std::shared_ptr<PythonAwaitWrapper>>(
+      m, "Await")
+      .def(
+          "wait",
+          &PythonAwaitWrapper::wait,
+          py::call_guard<py::gil_scoped_release>())
+      .def("fn", &PythonAwaitWrapper::fn)
+      .def("args", &PythonAwaitWrapper::args)
+      .def("type", &PythonAwaitWrapper::type)
+      .def("is_nowait", &PythonAwaitWrapper::is_nowait)
+      .def(
+          "__getattr__",
+          [](PythonAwaitWrapper& self, const std::string& name) -> py::object {
+            // LazyAwaitable semantic
+            return py::getattr(self.wait(), name.c_str(), py::none());
```

Comment:
remind me how this works for a scripted awaitable use case? (and maybe you can add this as a comment?)

(I know you explained this over video chat, but I forgot)

Reply:
For eager mode we allow to work with Await[W] as with W (automatically calling wait() if it was not called before).

For scripted awaitable we do not allow this - we expect explicit call `aw: Await[W]; W w = _awaitable_wait(aw)` and have explicit result w

Reply:
thanks, I think a description in a comment here would be good